step/src/main/java/com/google/sps/User.java

```diff
@@ -16,59 +16,65 @@
 
 import java.util.Collection;
 import java.util.HashSet;
+import java.util.ArrayList;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
 
 public class User {
   
-  private long id;
+  private String id;
   private String email;
   private String name;
   private Collection<User> matches;
+  
 
-  // TODO: determine what fields we want.
-  // TODO: update the constructor so that it only needs to take
-  //       in an id, and the other fields can then be looked up
-  //       in the database. this update will be made when data-
-  //       -store is integrated
-  public User(long id, String email, String name) {
+  public User(String id) {
     this.id = id;
-    this.email = email;
-    this.name = name;
-    //TODO: update this to fetch from database instead of making a new hashset    
-    this.matches = new HashSet<>();
   }
 
-  public long getId() {
+  // ID getter method
+  public String getId() {
     return id;    
   }
 
+  // Key getter method
+  public Key getKey() {
+    return KeyFactory.createKey("User", id);
```

Comment:
Does this createKey method return the Key if it already exists? Seems like duplicate calls to getKey() could be problematic.

Reply:
Yes it does. I was running into issues with testing when I was generating it once and storing it in a field but I just switched back and it worked this time. Hm